code/datums/gamemodes/combat_patrol.dm

```diff
@@ -0,0 +1,265 @@
+
+///round timer
+/datum/game_mode/combat_patrol/proc/set_game_timer()
+	if(!iscombatpatrolgamemode(SSticker.mode))
+		return
+	var/datum/game_mode/combat_patrol/D = SSticker.mode
+
+	if(D.game_timer)
+		return
+
+	D.game_timer = addtimer(CALLBACK(D, /datum/game_mode/combat_patrol.proc/set_game_end), max_game_time, TIMER_STOPPABLE)
+
+/datum/game_mode/combat_patrol/game_end_countdown()
+	if(!game_timer)
+		return
+	var/eta = timeleft(game_timer) * 0.1
+	if(eta > 0)
+		return "[(eta / 60) % 60]:[add_leading(num2text(eta % 60), 2, "0")]"
+	else
+		return "Patrol finished"
+
+/datum/game_mode/combat_patrol/proc/set_game_end()
+	max_time_reached = TRUE
+
+/////////
+/datum/game_mode/combat_patrol/process()
+	if(round_finished)
+		return PROCESS_KILL
+
+	if(TIMER_COOLDOWN_CHECK(src, COOLDOWN_BIOSCAN) || bioscan_interval == 0)
+		return
+	announce_bioscans_marine_som()
+
+// make sure you don't turn 0 into a false positive
+#define BIOSCAN_DELTA(count, delta) count ? max(0, count + rand(-delta, delta)) : 0
+
+///Annonce to everyone the number of xeno and marines on ship and ground
+/datum/game_mode/combat_patrol/proc/announce_bioscans_marine_som(show_locations = TRUE, delta = 2, announce_marines = TRUE, announce_som = TRUE)
+	TIMER_COOLDOWN_START(src, COOLDOWN_BIOSCAN, bioscan_interval)
+	var/list/list/counts = list(FACTION_TERRAGOV = 0, FACTION_SOM = 0)
+	var/list/list/area/locations = list(FACTION_TERRAGOV = 0, FACTION_SOM = 0)
+	var/list/list/som_list = GLOB.alive_human_list_faction[FACTION_SOM]
+	var/list/list/tgmc_list = GLOB.alive_human_list_faction[FACTION_TERRAGOV]
+
+	if(length(som_list))
+		counts[FACTION_SOM] += length(som_list)
+		locations[FACTION_SOM] = get_area(pick(som_list))
+	if(length(tgmc_list))
+		counts[FACTION_TERRAGOV] += length(tgmc_list)
+		locations[FACTION_TERRAGOV] = get_area(pick(tgmc_list))
+
+	//Adjust the randomness there so everyone gets the same thing
+	var/numTGMCr = BIOSCAN_DELTA(counts[FACTION_TERRAGOV], delta)
+	var/numSOMr = BIOSCAN_DELTA(counts[FACTION_SOM], delta)
+	var/TGMCLocation = locations[FACTION_TERRAGOV]
+	var/SOMLocation	= locations[FACTION_SOM]
+
+	//announcement for SOM
+	var/som_scan_name = "Long Range Tactical Bioscan Status"
+	var/som_scan_input = {"Bioscan complete.
+
+Sensors indicate [numTGMCr || "no"] unknown lifeform signature[numTGMCr > 1 ? "s":""] present in the area of operation[TGMCLocation ? ", including one at:[TGMCLocation]":""]"}
+
+	if(announce_som)
+		priority_announce(som_scan_input, som_scan_name, sound = 'sound/AI/bioscan.ogg', receivers = (som_list + GLOB.observer_list))
+
+	//announcement for TGMC
+	var/marine_scan_name = "Long Range Tactical Bioscan Status"
+	var/marine_scan_input = {"Bioscan complete.
+
+Sensors indicate [numSOMr || "no"] unknown lifeform signature[numSOMr > 1 ? "s":""] present in the area of operation[SOMLocation ? ", including one at:[SOMLocation]":""]"}
```

Comment:
```suggestion
Sensors indicate [numSOMr || "no"] unknown IFF signature[numSOMr > 1 ? "s":""] present in the area of operations[SOMLocation ? ", including one in [SOMLocation]":""]"}
```

Reply:
Good catch on operations. 
"At" is used else where and both make grammatical sense, so I'll leave it as is for consistancy. Fixed the missing space though.
Lifeform I also want to keep because it's generic, and consistant with bioscan in HvX because xenos sure don't have IFF. Also the name bioscan implies it's just picking up lifesigns somehow.